test/tests/functional/pbs_node_rampdown.py

```diff
@@ -6362,3 +6415,774 @@ def test_release_nodes_shared_server_restart_immed(self):
+                                 'Resource_List.ncpus': 7,
+                                 'Resource_List.select': newsel,
+                                 'Resource_List.place': self.job1_place,
+                                 'Resource_List.nodect': 3,
+                                 'schedselect': newsel,
+                                 'exec_host': self.job1_exec_host,
+                                 'exec_vnode': new_exec_vnode}, id=jid)
+
+    @timeout(500)
+    def test_release_multi_jobs(self):
+        """
+        Release vnodes when multiple jobs are present
+        """
+
+        # Delete the vnodes and recreate them
+        self.momA.delete_vnode_defs()
+        self.momB.delete_vnode_defs()
+        self.momA.restart()
+        self.momB.restart()
+        self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostC)
+
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 3},
+                            id=self.hostA)
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 3},
+                            id=self.hostB)
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 3},
+                            id=self.hostC)
+
+        # Submit multiple jobs
+        jid1 = self.create_and_submit_job('job13')
+        jid2 = self.create_and_submit_job('job13')
+        jid3 = self.create_and_submit_job('job13')
+
+        e_host_j1 = self.hostA + "/0+" + self.hostB + "/0+" + self.hostC + "/0"
+        e_host_j2 = self.hostA + "/1+" + self.hostB + "/1+" + self.hostC + "/1"
+        e_host_j3 = self.hostA + "/2+" + self.hostB + "/2+" + self.hostC + "/2"
+        e_vnode = "(%s:ncpus=1)+(%s:ncpus=1)+(%s:ncpus=1)" \
+            % (self.hostA, self.hostB, self.hostC)
+
+        self.server.expect(JOB, {"job_state=R": 3})
+        self.server.expect(JOB, {"exec_host": e_host_j1,
+                                 "exec_vnode": e_vnode}, id=jid1)
+        self.server.expect(JOB, {"exec_host": e_host_j2,
+                                 "exec_vnode": e_vnode}, id=jid2)
+        self.server.expect(JOB, {"exec_host": e_host_j3,
+                                 "exec_vnode": e_vnode}, id=jid3)
+
+        # Verify that 3 processes running on hostB
+        process = 0
+        self.server.pu.get_proc_info(
+            self.momB.hostname, ".*fib.*", None, regexp=True)
+        if (self.server.pu.processes is not None):
+            for key in self.server.pu.processes:
+                if ("fib" in key):
+                    process = len(self.server.pu.processes[key])
+                    print "length of the process is %d " % (process,)
```

Comment:
Don't use print, use self.logger.  I don't find this log message useful as it is.  I'd either add " (expected 3)" to the end or remove the message.  Same for the other use of process.

Reply:
sorry left as part of debugging. Will change.